src/class-drip-woocommerce-cart-events.php

```diff
@@ -118,7 +122,28 @@ private function remove_drip_cart_session_id()
     private function generate_drip_cart_session_id()
     {
         $random_data = random_bytes(32); // as of php7, random_bytes is advertised as cryptographically secure
-        return hash('sha256',  $random_data); 
+        return hash('sha256',  $random_data);
+    }
+
+    private function find_drip_visitor_uuid()
+    {
+        $account_id = WC_Admin_Settings::get_option( Drip_Woocommerce_Settings::ACCOUNT_ID_KEY );
+        if ($account_id === "") { return; }
+
+        $drip_cookie = urldecode($_COOKIE["_drip_client_{$account_id}"]);
```

Comment:
So if the cookie has not been set, for whatever reason, this could generate an undefined exception.  Maybe:
```
$drip_cookie = empty($_COOKIE["_drip_client_{$account_id}"]) ? "" : urldecode($_COOKIE["_drip_client_{$account_id}"]);
```

(or something less ugly)

Reply:
That's a good call. I'm going to add a test to make sure we handle the no-cookie case without error.

Reply:
I added a test, but could get an error/failure. I think maybe `urldecode` returns an empty string when `NULL` is passed in?